lib/common/memory.c

```diff
@@ -139,10 +139,12 @@ void h2o_mem_clear_pool(h2o_mem_pool_t *pool)
     pool->chunk_offset = sizeof(pool->chunks->bytes);
 }
 
-void *h2o_mem_alloc_pool(h2o_mem_pool_t *pool, size_t sz)
+void *h2o_mem_alloc_pool_aligned(h2o_mem_pool_t *pool, size_t alignment, size_t sz)
 {
     void *ret;
 
+    assert(alignment <= 2 * sizeof(void *) && sz % alignment == 0);
```

Comment:
Can we remove this assert?

I do not think that `alignment <= 2 * sizeof(void *)` is something that is guaranteed on every platform. `sz % alignment == 0` is going to be slow, and I do not think we need to test this considering how the function is used; it's essentially trying to detect a compiler bug in using the compiler itself.

Related to this, I think that we should eliminate `st_h2o_mem_pool_chunk_t::_dummy`. Instead, I would suggest changing the definition of `st_h2o_mem_pool_chunk_t` to something like below and start `h2o_mem_pool_t::chunk_offset` from `offsetof(union un_h2o_mem_pool_chunk_t, next)`.
```
union un_h2o_mem_pool_chunk_t {
    struct st_h2o_mem_pool_chunk_t *next;
    char bytes[4096];
};
```